components/utils.py

```diff
@@ -520,3 +606,282 @@ def untar(source, destination='/tmp', strip=1):
     ctx.logger.debug('Extracting {0} to {1}...'.format(source, destination))
     sudo(['tar', '-xzvf', source, '-C', destination,
           '--strip={0}'.format(strip)])
+
+
+def write_file(content, file_path):
+    tmp_file = tempfile.NamedTemporaryFile(delete=False)
+    mkdir(os.path.dirname(os.path.abspath(file_path)))
```

Comment:
I wouldn't add that here. this function explicitly writes a file.. it's called write_file. I would create a directory before running the write_file outside of this function.